libs/@blockprotocol/type-system/rust/src/schema/array/validation.rs

```diff
@@ -0,0 +1,29 @@
+use thiserror::Error;
+
+use crate::{Valid, Validator, schema::PropertyValueArray};
+
+#[derive(Debug, Error)]
+pub enum ArraySchemaValidationError {
+    #[error(
+        "Unsatisfiable item number constraints, expected minimum of {min} and maximum of {max} \
+         items"
+    )]
+    IncompatibleItemNumberConstraints { min: usize, max: usize },
+}
+
+pub struct ArraySchemaValidator;
+
+impl<T: Sync> Validator<PropertyValueArray<T>> for ArraySchemaValidator {
+    type Error = ArraySchemaValidationError;
+
+    fn validate_ref<'v>(
+        &self,
+        value: &'v PropertyValueArray<T>,
+    ) -> Result<&'v Valid<PropertyValueArray<T>>, Self::Error> {
+        if let Some((min, max)) = value.min_items.zip(value.max_items) {
```

Comment:
I'm not understanding how this achieves the description in the PR:

> This PR adds a validation for array property types to ensure that minItems must be less than or equal to maxItems.

Does it happen elsewhere / am I missing something? 

And do we already do this for min/max in an entity type's links schema?

Reply:
This validates the array property ***types***. If a property type expects an array with a `minItems` and `maxItems` value it would not be possible to instantiate a property for it if `minItems > `maxItems`.

e.g. this property type would now fail:
```json
{
  "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/property-type",
  "kind": "propertyType",
  "$id": "https://blockprotocol.org/@alice/types/property-type/contrived-property/v/1",
  "title": "Contrived Property",
  "description": "A contrived property for testing purposes.",
  "oneOf": [
    {
      "type": "array",
      "items": {
        "oneOf": [
          {
            "$ref": "https://blockprotocol.org/@blockprotocol/types/data-type/number/v/1"
          }
        ]
      },
      "minItems": 4
      "maxItems": 2
    }
  ]
}
```

Reply:
> And do we already do this for min/max in an entity type's links schema?

Good flag, no, we don't

Reply:
I added the proper check, improved the error message, and added link count validation as well: [`7565585` (#5827)](https://github.com/hashintel/hash/pull/5827/commits/75655855d2ff37f1b8c9de732fc58eb5a17601db)

Reply:
What about if `max < min`?

Reply:
`max < min` is the same as `min > max` I think 🙂

Reply:
Oh yeah :D